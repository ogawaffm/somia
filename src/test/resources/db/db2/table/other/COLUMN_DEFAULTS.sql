DROP TABLE IF EXISTS TEST.COLUMN_DEFAULTS
/
CREATE TABLE TEST.COLUMN_DEFAULTS
(
    VARCHAR_10_DEFAULT_TEXT_COL             VARCHAR(10) DEFAULT 'TEXT',
    DECIMAL_10_2_DEFAULT_1_23_COL           DECIMAL(10, 2) DEFAULT 1.23,
    TIMESTAMP_DEFAULT_CURRENT_TIMESTAMP_COL TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    BOOLEAN_DEFAULT_FALSE_COL               BOOLEAN DEFAULT FALSE,
    "BOOLEAN_DEFAULT_true_COL"              BOOLEAN DEFAULT true,
    INTEGER_DEFAULT_NULL_COL                INTEGER DEFAULT NULL
)