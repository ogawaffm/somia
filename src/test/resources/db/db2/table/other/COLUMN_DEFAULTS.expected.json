{
  "name": "COLUMN_DEFAULTS",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "VARCHAR_10_DEFAULT_TEXT_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 10,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 10,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": "'TEXT'"
    },
    {
      "name": "DECIMAL_10_2_DEFAULT_1_23_COL",
      "type": {
        "name": "DECIMAL",
        "typeNumber": 3,
        "className": "java.math.BigDecimal",
        "precision": 10,
        "scale": 2,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 10,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": "1.23"
    },
    {
      "name": "TIMESTAMP_DEFAULT_CURRENT_TIMESTAMP_COL",
      "type": {
        "name": "TIMESTAMP",
        "typeNumber": 93,
        "className": "java.sql.Timestamp",
        "precision": 26,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 46,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": "CURRENT TIMESTAMP"
    },
    {
      "name": "BOOLEAN_DEFAULT_FALSE_COL",
      "type": {
        "name": "BOOLEAN",
        "typeNumber": 16,
        "className": "java.lang.Boolean",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 5,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": "FALSE"
    },
    {
      "name": "BOOLEAN_DEFAULT_true_COL",
      "type": {
        "name": "BOOLEAN",
        "typeNumber": 16,
        "className": "java.lang.Boolean",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 5,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": "true"
    },
    {
      "name": "INTEGER_DEFAULT_NULL_COL",
      "type": {
        "name": "INTEGER",
        "typeNumber": 4,
        "className": "java.lang.Integer",
        "precision": 10,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 11,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": "NULL"
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": false,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}